zmq.hpp

```diff
@@ -2224,6 +2224,53 @@ class monitor_t
         on_monitor_started();
     }
 
+#if ZMQ_VERSION_MAJOR >= 4
+    bool get_event(zmq_event_t& eventMsg, std::string& address, zmq::recv_flags flags = zmq::recv_flags::none)
+    {
+        assert(_monitor_socket);
+
+        eventMsg.event = 0;
+        eventMsg.value = 0;
+        address = std::string();
+
+        {
+            message_t msg;
+            int rc = zmq_msg_recv(msg.handle(), _monitor_socket.handle(),
+                static_cast<int>(flags));
+
+            if (rc == -1)
+            {
+                if (zmq_errno() == ETERM || zmq_errno() == EAGAIN)
+                    return false;
```

Comment:
Why does this also returns false on `ETERM`? There's no way to distinguish `EAGAIN` and `ETERM` this way, and this seems inconsistent with the behavior of `zmq::socket_t::recv`.

Reply:
Check https://github.com/zeromq/cppzmq/blob/master/zmq.hpp#L2274.

I just kept same behaviour implemented in check_event(). If this is not mandatory, I can just throw an exception and not use  a std::optionnal as described above, but just std::pair<>

Reply:
Hm, yes, that's right. It's inconsistent one way or the other, but I think the behaviour of `zmq::socket_t::recv` is the appropriate one.

Reply:
what we should implement in this case? Maybe we shall throw an exception?